### ISSUE TITLE
Is this still a thing' ? I mean were on DirectX 12 now..

### ISSUE THREAD

**Issue by WilsonBradley** [fholger/vrperfkit#156]
Would be great to get some TLC / love for new revision..

**Comment by WilsonBradley** [comment]
New Nvidia drivers/settings And RTX 40' series

**Comment by fholger** [comment]
Games that are new enough to use DX12 should be new enough to use OpenXR. In which case you should be looking at the OpenXR toolkit and not this.

**Comment by ThickCloud** [comment]
In my limited experience, this solution provides more flexibility, a superior capability set, simpler implementation and, most importantly, better performance that openxr...

I switched back from OpenXR last summer/fall.

Brilliant work!

**Comment by oskansavli** [comment]
I agree that SteamVR + vrperfkit is faster than OpenXR with the integrated FFR tool. Tested multiple times with RTX4090 and RTX3070 mobile. OpenXR uses a bit less VRAM though.

I hope this is updated for DX12.

**Comment by Zod1234** [comment]
does this have all the features of fsr2?  I wanted to see how dynamic resolution scaling performed.   just a side note, the predecessor of vrperfkit still has uses.  I recently started playing valheim which is a vrmod that uses the same DLL so to get any resolution scaling I was your prior work existed.  it seems this would still potentially be an issue for other vr mods.

**Comment by fholger** [comment]
No, this is FSR1, not 2. And it never will be 2 - you can't integrate FSR2 into games in a generic way as it requires closer engine integration.

**Comment by theguycanfly** [comment]
I don't believe this is of any use anymore as far 1 is built into the and
driver at this point

**Comment by fholger** [comment]
Driver features do not apply to VR.

**Comment by theguycanfly** [comment]
Ah true

**Comment by Zod1234** [comment]
> No, this is FSR1, not 2. And it never will be 2 - you can't integrate FSR2 into games in a generic way as it requires closer engine integration.

you're right of course.  did some more reading and makes sense because it's a temporal scaler.  but did more reading, amd also has something called Radeon boost which is dynamic resolution scaling.   I feel dynamic resolution scaling could be more impactful because you can be more aggressive to hit your fps target and when the action slows down, it'll scale up.  I read this is also only per game currently, but I couldn't find out if there is a technical barrier that would prevent this from being applied to any game.  one benefit of doing it in game is avoiding doing it on the HUD, but besides that, if there's no temporal info needed, couldn't this in theory work in any game?  Is it in their open source release?

I've never tried dynamic res in a game but curious how well it would work.  not many pc games have done this prior to Radeon boost.  One that did is doom 2016, but strangely it was disabled on PC and only used in consoles.  Here's a vid of it being unlocked on PC and it looks pretty impressive.  this looks like intelligent scaling trying to reach an fps target rather than the static scaling we use today.

Did you have any thoughts on openvr_fsr still having use?  I know the future is openxr, but the transition is taking a long time.  And again with mods helping vr a fair bit, openvr_fsr is the only option when mods are used.  still works fine as is, just thinking if vrperfkit did get new features that potentially openvr_perf could benefit from.  Although it sounds like both are considered feature complete at this point.  read you paused to work on hl2.  just finished that recently.  absolutely awesome in vr!  thx for all your contributions.

https://www.youtube.com/watch?v=GS_PhWP9YQI&t=746s

**Comment by Zod1234** [comment]
> Games that are new enough to use DX12 should be new enough to use OpenXR. In which case you should be looking at the OpenXR toolkit and not this.

losing out on steamvr bindings is tough.   and I much prefer this method of deployment with a custom cfg tailored to the game VS an onscreen display.  ideally we would all be using openxr, but the reality is what's keeping pcvr alive is mods and very few are using openxr.  and those that do seem to have trouble with openxr toolkit.  just tried dx12 re2 vr mod set to openxr with the toolkit and it locked my system.

I get your sentiment but I think many would benefit from dx12 support in vrperfkit.  i know I would.
